lib/html/proofer/checkable.rb

```diff
@@ -30,7 +30,11 @@ def initialize(obj, check)
       end
 
       def url
-        @src || @srcset || @href || ''
+        url = @src || @srcset || @href || ''
+        if @check.base
```

Comment:
For the tests, this can change to

``` ruby
if @check && @check.base
```

The reason `@check` is `nil` is because of the tests in *spec/checkable_spec.rb*. Technically you can parse HTML without associating it with a check to run.